### PR TITLE
Add unit tests for aspect package

### PR DESCRIPTION
**Pull request by wwelling**

### CODE REVIEW

File: src/test/java/org/folio/rest/camunda/aspect/TenantInjectionDelegateAspectTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  private DelegateTask task;
+
+  @Mock
+  private TenantProperties tenantProperties;
+
+  @InjectMocks
+  private TenantInjectionDelegateAspect tenantInjectionDelegateAspect;
+
+  @BeforeEach
+  void mockDefaultTenant() {
```
[the remark this answers is not in the document]
Comment (wwelling) [reply src/test/java/org/folio/rest/camunda/aspect/TenantInjectionDelegateAspectTest.java:36]: Why? Doesn't the annotation indicate that and not what the method actually does. I would think using `beforeEach` as a method name would be less useful here. Is this more for a matter of consistency and refactoring?

File: src/test/java/org/folio/rest/camunda/aspect/TenantInjectionDelegateAspectTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  @Test
+  void testBeforeTaskListenerNotify() {
+    tenantInjectionDelegateAspect.beforeTaskListenerNotify(task);
+
+    Mockito.verify(task).getTenantId();
+
+
+    lenient().when(execution.getTenantId()).thenReturn(null);
+    lenient().when(task.getTenantId()).thenReturn(null);
+    lenient().when(tenantProperties.isForceTenant()).thenReturn(true);
```
[the remark this answers is not in the document]
Comment (wwelling) [reply src/test/java/org/folio/rest/camunda/aspect/TenantInjectionDelegateAspectTest.java:76]: Not committing changes as potential to be a breaking commit.

File: src/test/java/org/folio/rest/camunda/aspect/TenantInjectionDelegateAspectTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+  @Test
+  void testBeforeExecutionListenerNotify() {
+    tenantInjectionDelegateAspect.beforeExecutionListenerNotify(execution);
+
+    Mockito.verify(execution).getTenantId();
+
+
+    lenient().when(execution.getTenantId()).thenReturn(null);
+    lenient().when(tenantProperties.isForceTenant()).thenReturn(true);
```
[the remark this answers is not in the document]
Comment (wwelling) [reply src/test/java/org/folio/rest/camunda/aspect/TenantInjectionDelegateAspectTest.java:63]: Not committing changes as potential to be a breaking commit.

File: src/test/java/org/folio/rest/camunda/aspect/TenantInjectionDelegateAspectTest.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+  @Test
+  void testBeforeDelegateExecution() {
+    tenantInjectionDelegateAspect.beforeDelegateExecution(execution);
+
+    Mockito.verify(execution).getTenantId();
+
+
+    lenient().when(execution.getTenantId()).thenReturn(null);
+    lenient().when(tenantProperties.isForceTenant()).thenReturn(true);
```
[the remark this answers is not in the document]
Comment (wwelling) [reply src/test/java/org/folio/rest/camunda/aspect/TenantInjectionDelegateAspectTest.java:51]: Not committing changes as potential to be a breaking commit.

File: src/test/java/org/folio/rest/camunda/aspect/TenantInjectionDelegateAspectTest.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+  @InjectMocks
+  private TenantInjectionDelegateAspect tenantInjectionDelegateAspect;
+
+  @BeforeEach
+  void mockDefaultTenant() {
+    lenient().when(execution.getTenantId()).thenReturn("diku");
+    lenient().when(task.getTenantId()).thenReturn("diku");
+    lenient().when(tenantProperties.getDefaultTenant()).thenReturn("diku");
+    lenient().when(tenantProperties.isForceTenant()).thenReturn(false);
```
[the remark this answers is not in the document]
Comment (wwelling) [reply src/test/java/org/folio/rest/camunda/aspect/TenantInjectionDelegateAspectTest.java:40]: Not committing changes as potential to be a breaking commit.

File: src/test/java/org/folio/rest/camunda/aspect/TenantInjectionDelegateAspectTest.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
+package org.folio.rest.camunda.aspect;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.lenient;
```
[the remark this answers is not in the document]
Comment (wwelling) [reply src/test/java/org/folio/rest/camunda/aspect/TenantInjectionDelegateAspectTest.java:4]: Not committing changes as potential to be a breaking commit.

**Comment by wwelling** [comment]
Refactored before each method and removed unnecessary stubbing.

File: src/test/java/org/folio/rest/camunda/aspect/TenantInjectionDelegateAspectTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaladay) [reply src/test/java/org/folio/rest/camunda/aspect/TenantInjectionDelegateAspectTest.java]: We should be generalized with the `beforeEach` and similar.
There should only be one and we should be able to easily and consistently find it.
Such functions can also have their scope expand as more changes happen over time.

**Review by kaladay — APPROVED** [review]

**Pull request merged**